acme/src/main/java/io/micronaut/acme/services/AcmeService.java

```diff
@@ -433,7 +433,7 @@ private void doChallengeSpecificSetup(Authorization auth, Challenge challenge) t
                 eventPublisher.publishEvent(new HttpChallengeDetails(http01Challenge.getToken(), http01Challenge.getAuthorization()));
 
                 // Configuring self signed until a real cert is available.
-                SelfSignedCertificate ssc = new SelfSignedCertificate(acmeConfiguration.getDomain());
+                SelfSignedCertificate ssc = new SelfSignedCertificate(acmeConfiguration.getDomains().stream().findFirst().get());
```

Comment:
Just sanity checking, it's safe for the self signed cert to always be for the first domain?

Reply:
I don't think a domain there is even necessary. The micronaut implementation for the server just uses the no arg constructor

Reply:
Removed completely in 7ea17de73ee257b05563491b49a148bb8ed1ab8f. Instead of putting a dummy self signed cert in place that basically gets you a half working app server. I removed it and instead now will stop the application from starting if there is an issue in setting up the certificate.